packages/roosterjs-editor-api/lib/utils/processList.ts

```diff
@@ -5,65 +5,65 @@ import {
     isVoidHtmlElement,
     isBlockElement,
     Browser,
-    isNodeEmpty,
-    getTagOfNode,
-    getComputedStyles,
-    applyFormat,
-    getFirstLeafNode,
 } from 'roosterjs-editor-dom';
 
 const TEMP_NODE_CLASS = 'ROOSTERJS_TEMP_NODE_FOR_LIST';
-const TEMP_NODE_HTML = `<img class="${TEMP_NODE_CLASS}">`;
+const TEMP_NODE_HTML = "<img class=\"" + TEMP_NODE_CLASS + "\">";
+const SELECTION_NODE_CLASS = 'ROOSTERJS_SELECTION_NODE';
+const SELECTION_NODE_HTML = "<img class=\"" + SELECTION_NODE_CLASS + "\">";
+
+type ValidProcessListDocumentCommands = DocumentCommand.Outdent | DocumentCommand.Indent | DocumentCommand.InsertOrderedList | DocumentCommand.InsertUnorderedList;
 
 /**
  * Browsers don't handle bullet/numbering list well, especially the formats when switching list statue
  * So we workaround it by always adding format to list element
  */
-export default function processList(editor: Editor, command: DocumentCommand): Node {
-    if (Browser.isChrome) {
+export default function processList(editor: Editor, command: ValidProcessListDocumentCommands): Node {
+    let currentNode = editor.getElementAtCursor();

```

Comment:
Should we get current LI node instead?

Reply:
Yeah, that makes sense especially for complex structures under LI nodes.